### ISSUE TITLE
not able to decrypt keys in bootstrap.properties

### ISSUE THREAD

**Issue by nagarajatantry** [ulisesbocchio/jasypt-spring-boot#63]
Hi,

I am using jasypt 1.15 version and spring boot 1.5.7 version. In my Spring cloud config client, I have below configuration in bootstrap.properties

```
spring.application.name=myapp
spring.cloud.config.uri=http://localhost:8888
spring.cloud.config.username=admin
spring.cloud.config.password=ENC(encrypted value)
```

I have the below code in the main method.
```
System.setProperty("jasypt.encryptor.password","key");
new SpringApplicationBuilder()
	.environment(new StandardEncryptableEnvironment())
	.listeners(new ApplicationPidFileWriter("application.pid"))
	.sources(ConfigApp.class)
	.run(args);

```
Looks like jasypt is not able to decrypt 'spring.cloud.config.password'. I get a 401 error from my spring cloud server. If i replace the password with plain text, then it works.
Can someone please take a look at this?

**Comment by Pauki** [comment]
Hi, 

I hit the same issue as you and I figure it out adding spring.factories to my app META-INF directory with:
org.springframework.cloud.bootstrap.BootstrapConfiguration = com.ulisesbocchio.jasyptspringboot.JasyptSpringBootAutoConfiguration

**Comment by ulisesbocchio** [comment]
that's weird @Pauki, using the `jasypt-spring-boot-starter` dependency does exactly that. It's an empty `jar` with one file in META-INF with that exact line of configuration.
When you're adding that config explicitly you're basically adding config twice, once by the `StandardEncryptableEnvironment` and another one with the auto config. Both configs check for the existence of encryptable properties sources before acting, so it's safe to put both, but kinda overkill. 
bootstrap.properties is special though. My only guess is that somehow bootstrap.properties is going outside of the environment, but let me dig on it.

**Comment by wwk3j** [comment]
I am starting to use the jasypt-spring-boot package and I was wondering whether it works with org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider? If not I may have to abandon it. I originally attempted to use the jasypt spring 3.1 integration with encryptors as beans but that didn't quite work with bc-fips as a provider.

**Comment by ulisesbocchio** [comment]
this is fixed in 1.17

**Comment by nav16011991** [comment]
I am using spring cloud config server ( version : Greenwich.SR1) with jasypt which is working fine. But when i use this server to fetch configuration for other service (with jasypt-spring-boot-starter:2.1.0), It throw following ERROR (not exception)
java.lang.AbstractMethodError
	at org.springframework.core.OrderComparator.findOrder(OrderComparator.java:142)
	at org.springframework.core.annotation.AnnotationAwareOrderComparator.findOrder(AnnotationAwareOrderComparator.java:65)
	at org.springframework.core.OrderComparator.getOrder(OrderComparator.java:125)
	at org.springframework.core.OrderComparator.getOrder(OrderComparator.java:113)
	at org.springframework.core.OrderComparator.doCompare(OrderComparator.java:82)
	at org.springframework.core.OrderComparator.compare(OrderComparator.java:68)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:360)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1462)
	at org.springframework.boot.SpringApplication.asUnmodifiableOrderedSet(SpringApplication.java:1344)
	at org.springframework.boot.SpringApplication.getListeners(SpringApplication.java:1253)
	at org.springframework.boot.context.event.EventPublishingRunListener.contextLoaded(EventPublishingRunListener.java:85)
	at org.springframework.boot.SpringApplicationRunListeners.contextLoaded(SpringApplicationRunListeners.java:66)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:408)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:331)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1277)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1265)
